streaming/base/format/mds/encodings.py

```diff
@@ -510,6 +510,8 @@ class JSON(Encoding):
     """Store arbitrary data as JSON."""
 
     def encode(self, obj: Any) -> bytes:
+        if isinstance(obj, np.ndarray):
```

Comment:
hm... i agree that this makes sense, but why would we encode ndarrays as json cols instead of directly using the ndarray encoding types that we support?

Reply:
I don't think that generates JSON though. The use case I have in mind (which I should have spelled out) is generating chat-style datasets.

Each input is an array of messages like `{'role': 'user', 'content': '...'}` and we want to generate a JSON array of JSON dicts. So the `'json'` type does seem most natural; this isn't an array of numbers for example and the nested elements do have to be JSON-encoded.

(This relates to my other PR, where we also need support for map types to make those dicts work)

Reply:
I see. thanks